src/test/java/io/r2dbc/postgresql/client/ReactorNettyClientTest.java

```diff
@@ -205,7 +205,141 @@ private PostgresqlConnectionFactory createConnectionFactory(String username, Str
                 .applicationName(ReactorNettyClientTest.class.getName())
                 .build());
         }
+    }
+
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    final class SslTest {
+
+        private ReactorNettyClient client;
+
+        @AfterEach
+        void closeClient() {
+            if (this.client != null) {
+                this.client.close().block();
+            }
+            this.client = null;
+        }
+
+
+        @Test
+        void exchangeSslWithPassword() throws IOException {
+            SslProvider sslProvider = SslProvider.builder()
+                .sslContext(SslContextBuilder.forClient()
+                    .trustManager(SERVER.getServerCrt().getInputStream())
+                )
+                .defaultConfiguration(TCP)
+                .build();
+
+            this.client = ReactorNettyClient.connect(SERVER.getHost(), SERVER.getPort(), sslProvider, null)
```

Comment:
Having all these `.block()` methods in the code is an anti-pattern as they don't time out. We should use at least `StepVerifier` where possible.